### PR TITLE
[T4A3][T13-B2] Gabriel Gheorghian

### PR DESCRIPTION
**Pull request by GabrielGhe**
I branched off the last activity unfortunately.
Ready for review.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,53 +6,21 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
```
Comment (KulaniT) [inline src/seedu/addressbook/data/person/Address.java:14]: why public?

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -96,7 +96,7 @@ public Command parseCommand(String userInput) {
      * @param args full command args string
      * @return the prepared command
      */
-    private Command prepareAdd(String args){
+    private static Command prepareAdd(String args){
```
Comment (KulaniT) [inline src/seedu/addressbook/parser/Parser.java:99]: The purpose here is to understand when you need to have a class level method. So do you think all the methods in Parser class need to be class-level? Think about it.

**Comment by KulaniT** [comment]
@GabrielGhe : overall good. Practice to give the lowest possible access modifier.

**Comment by KulaniT** [comment]
@GabrielGhe : Some comments added. Please close the PR after reading comments

File: src/seedu/addressbook/parser/Parser.java (hunk 2 above)
Comment (GabrielGhe) [reply src/seedu/addressbook/parser/Parser.java:99]: You can use a class level method when you don't necessarily need an instance of the object. Like StringUtils or Parser.

File: src/seedu/addressbook/data/person/Address.java (hunk 1 above)
Comment (GabrielGhe) [reply src/seedu/addressbook/data/person/Address.java:14]: For tests. Normally you'd have something like this as visible for tests using guava.

**Pull request closed without merging**